src/main/java/org/folio/service/uc/export/ExportServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package org.folio.service.uc.export;
+
+import static com.opencsv.ICSVWriter.DEFAULT_LINE_END;
+import static com.opencsv.ICSVWriter.NO_ESCAPE_CHARACTER;
+import static com.opencsv.ICSVWriter.NO_QUOTE_CHARACTER;
+
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import com.opencsv.bean.StatefulBeanToCsv;
+import com.opencsv.bean.StatefulBeanToCsvBuilder;
+import com.opencsv.exceptions.CsvDataTypeMismatchException;
+import com.opencsv.exceptions.CsvRequiredFieldEmptyException;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import org.folio.rest.converter.costperuse.export.PackageTitlesCollectionConverter;
+import org.folio.service.uc.UCCostPerUseService;
+
+@Service
+public class ExportServiceImpl implements ExportService {
+  private static final Logger LOG = LoggerFactory.getLogger(ExportServiceImpl.class);
+
+  @Autowired
+  private UCCostPerUseService costPerUseService;
+  @Autowired
+  private PackageTitlesCollectionConverter converter;
+
+  public CompletableFuture<String> exportCSV(String packageId, String platform, String year,
+                                             Map<String, String> okapiHeaders) {
+    LOG.info("Perform export for package - " + packageId);
+    return costPerUseService.getPackageResourcesCostPerUse(packageId, platform, year, okapiHeaders)
+        .thenApply(collection -> converter.convert(collection, platform, year))
+        .thenCompose(this::mapToCSV);
+  }
+
+  private CompletableFuture<String> mapToCSV(List<TitleExportModel> entities) {
+    CompletableFuture<String> result = new CompletableFuture<>();
+    LOG.info("Mapping " + entities.size() + " entities to SCV");
+    StringWriter writer = new StringWriter();
+
+    // mapping of columns by position
+    CustomBeanToCSVMappingStrategy<TitleExportModel> mappingStrategy = new CustomBeanToCSVMappingStrategy<>();
+    mappingStrategy.setType(TitleExportModel.class);
+
+    StatefulBeanToCsv<TitleExportModel> csvToBean = new StatefulBeanToCsvBuilder<TitleExportModel>(writer)
+      .withMappingStrategy(mappingStrategy)
+      .withQuotechar(NO_QUOTE_CHARACTER)
+      .withEscapechar(NO_ESCAPE_CHARACTER)
+      .withLineEnd(DEFAULT_LINE_END)
+      .withSeparator('|')
```

Comment:
Why we use '|' this char as a separator? I think it is not a commonly used separator for CSV files.

Reply:
we use "|" character to align with the same separator as EBSCO uses, just not to confuse users. We can change the separator to any other character we agreed.

Reply:
As I remember in the example that Khalilla showed us this separator was used for data in one column. For example for ISBN column if there several ISBN's for title